### PR TITLE
CM-40771 - Code refactoring of CLI communication

### PR DESCRIPTION
**Pull request by MarshalX**
Now CLI communication aligns with our other plugins with proper typing 🎉

### CODE REVIEW

**Review by elsapet — APPROVED** [review]

**Comment by elsapet** [comment]
Concerning the SAST violations, we can mark the "Observable Timing Discrepancy" as false positives here. And if there is no external or user input involved, then the "Unsanitized dynamic input in OS command" can also be safely ignored.

File: src/providers/tree-data/utils.ts [hunk 1]
```diff
@@ -37,89 +45,83 @@ export const refreshTreeViewData = (
...
+  scanType: CliScanType,
 ): Map<string, TreeDisplayedData[]> => {
-  const resultMap: Map<string, TreeDisplayedData[]> = new Map();
+  const resultMap = new Map<string, TreeDisplayedData[]>();
 
   detections.forEach((detection) => {
     let valueItem: ValueItem | undefined;
 
-    if (scanType === ScanType.Secrets) {
+    if (scanType === CliScanType.Secret) {
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply src/providers/tree-data/utils.ts:118]: #cycode_sast_false_positive

File: src/ui/panels/violation/content.ts [hunk 2]
```diff
@@ -17,10 +17,10 @@ export default (detectionType: ScanType) => `
 <body>
     ${style}
 
-    ${detectionType == ScanType.Sca ? scaCard : ''}
-    ${detectionType == ScanType.Secrets ? secretCard : ''}
-    ${detectionType == ScanType.Iac ? iacCard : ''}
-    ${detectionType == ScanType.Sast ? sastCard : ''}
+    ${detectionType == CliScanType.Sca ? scaCard : ''}
+    ${detectionType == CliScanType.Secret ? secretCard : ''}
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply src/ui/panels/violation/content.ts:21]: #cycode_sast_false_positive

File: src/cli/cli-wrapper.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  public executeCommand<T extends ClassConstructor<unknown>>(
+    classConst: T | null, args: string[], cancellationToken?: vscode.CancellationToken,
+  ): Promise<CliResult<T>> {
+    const defaultCliArgs = [CommandParameters.OutputFormatJson, getUserAgentArg()];
+    const commandParams = [...config.additionalParams, ...defaultCliArgs, ...args];
+
+    const executedCommand = `${config.cliPath} ${commandParams.join(' ')}`;
+    this.logger.debug(`Running command: "${executedCommand}"`);
+
+    const childProcess = spawn(config.cliPath, commandParams, {
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply src/cli/cli-wrapper.ts:66]: #cycode_sast_ignore_here

**Pull request merged**